kie-pmml-trusty/kie-pmml-commons/src/main/java/org/kie/pmml/commons/model/predicates/KiePMMLCompoundPredicate.java

```
@@ -0,0 +1,139 @@
+        return booleanOperator == that.booleanOperator &&
+                Objects.equals(operatorFunction, that.operatorFunction) &&
+                Objects.equals(kiePMMLPredicates, that.kiePMMLPredicates);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), booleanOperator, operatorFunction, kiePMMLPredicates);
+    }
+
+    public static class Builder extends KiePMMLPredicate.Builder<KiePMMLCompoundPredicate> {
+
+        private Builder(List<KiePMMLExtension> extensions, BOOLEAN_OPERATOR booleanOperator) {
+            super("CompoundPredicate-", () -> new KiePMMLCompoundPredicate("CompoundPredicate", extensions, booleanOperator));
+            toBuild.operatorFunction = getInnerBinaryOperator(booleanOperator);
+        }
+
+        public KiePMMLCompoundPredicate.Builder withKiePMMLPredicates(List<KiePMMLPredicate> kiePMMLPredicates) {
+            kiePMMLPredicates.forEach(predicate -> predicate.setParentId(toBuild.id));
+            toBuild.kiePMMLPredicates = kiePMMLPredicates;
+            return this;
+        }
+
+        private BinaryOperator<Boolean> getInnerBinaryOperator(BOOLEAN_OPERATOR booleanOperator) {
+            switch (booleanOperator) {
+                // logic here is
+                // first boolean may be null (initial evaluation) so we start taking the second boolean
+                case OR:
+                    return (aBoolean, aBoolean2) -> aBoolean != null ? aBoolean || aBoolean2 : aBoolean2;
+                case AND:
+                    return (aBoolean, aBoolean2) -> aBoolean != null ? aBoolean && aBoolean2 : aBoolean2;
+                case XOR:
+                    return (aBoolean, aBoolean2) -> aBoolean != null ? aBoolean ^ aBoolean2 : aBoolean2;
+                case SURROGATE:
+                    // TODO {gcardosi} DROOLS-5594
+                default:
+                    return (aBoolean, aBoolean2) -> aBoolean;
```

Is there something like default binary operator? Shouldn't we rather throw an exception.

---

@jiripetrlik 
Removed - throwing exceptions